fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/Repository.java

```diff
@@ -133,6 +143,75 @@ public BinaryObject createBinary(URI path, String slug, InputStream content, Str
 		BinaryObject binary = new BinaryObject(newPid, this, repositoryObjectDataLoader);
 		return binary;
 	}
+	
+	/**
+	 * Creates an event for the specified object.
+	 * 
+	 * @param eventPid
+	 *            the PID of the event to add
+	 * @param model
+	 *            Model containing properties of this event. Must only contain
+	 *            the properties for one event.
+	 * @return URI of the event created
+	 * @throws FedoraException
+	 */
+	public PremisEventObject createPremisEvent(PID eventPid, Model model) throws FedoraException {
+
+		try (FcrepoResponse response = getClient().put(eventPid.getRepositoryUri())
+				.body(RDFModelUtil.streamModel(model), "text/turtle")
+				.perform()) {
+
+			return new PremisEventObject(PIDs.get(response.getLocation()), this, repositoryObjectDataLoader);
+		} catch (IOException e) {
+			throw new FedoraException("Unable to create premis event for " + eventPid, e);
+		} catch (FcrepoOperationFailedException e) {
+			throw ClientFaultResolver.resolve(e);
```

Comment:
What's going on behind the scenes here? I'm guessing resolve() returns a throwable, but is it doing anything else?